### DISCUSSION TITLE
OpenAI assistant with file retrieval only works if at least one file is added in Librechat

### DISCUSSION THREAD

**Discussion by avanwouwe** [danny-avila/LibreChat#2924] [Troubleshooting]
### What happened?

I'm trying to deploy Librechat. It is working great for many of my use cases, but one of them is stil resisting : an assistant that uses a vector store with ~450 files.

I have created an assistant in open AI, added a vector store with files, and it works in the open AI playground. But when I use that assistant in Librechat it doesn't use the vector index (no "Used retrieval" indicator). 

Only when I add (at least) one file to the assistant in the Librechat GUI, it will trigger file retrieval when querying in Librechat.  

However it results in results where it first states "It seems that the document I opened does not contain information about xxx", then performs a second retrieval, and then finds it.

Theoretically I could add all files via Librechat. But adding 450 files through Librechat is complicated; I augmented the rate limits, but it overwhelms the server and fails all upload requests.

Is there some way to get file retrieval to work without having to add the files via Librechat?

### Steps to Reproduce

1. create an assistant in Librechat (that uses file retrieval)
2. add many files in OpenAI (and none in Librechat)
3. ask a question about the documents

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Yes you don't need to add files on librechat beforehand. I'm not sure at what point it's not working for you, but this is how I get it to work. See my shared chat to see that it's in fact working.

https://demo.librechat.ai/share/3140f25b-b6a7-4fb3-90d8-5d8d6b393355

## Step 1: Create/Save Assistant on LibreChat, with File Search enabled

![image](https://github.com/danny-avila/LibreChat/assets/110412045/e8e3c7ef-a4bc-4203-8a77-87918a33cf5d)

whether or not file search is enabled at this step shouldn't matter, but I did so for good measure.

--- 

## Step 2: Create a vector store and/or add the files

![image](https://github.com/danny-avila/LibreChat/assets/110412045/09a70fb5-a428-429a-bad2-c95a5ae843c3)

Curiously, I no longer see the option to add files to this vector store, I'm not sure why. But in any case, I would make sure you see the files on this screen, and detach/attach to assistant to confirm it's used by your assistant

---

## Step 3: trigger a knowledge response

As you can see in my example, i just had to ask a general question. but forcing a retrieval may not be possible without "binding" which is not yet implemented and will have to look into that as a feature overall

**Comment by danny-avila** [comment]
Also FYI I've already drafted a lot of the UI for handling vector stores/files from LibreChat, it still needs more work, but this is coming.
